models/model.py

```diff
@@ -56,24 +69,35 @@ def training_step(self, batch, batch_idx):
             batch_idx     - Index of the batch in the dataset (not needed here).
         """
         # "batch" is the output of the training data loader
-        predictions, labels = self.model(batch, mode='train')
+        out, labels = self.model(batch, mode='train')
+        predictions = self.classifier(out)
         loss = self.loss_module(predictions, labels)
 
         self.log('train_accuracy', self.accuracy(predictions, labels).item(), on_step=False, on_epoch=True)
         self.log('train_loss', loss)
 
         # logging in optimizer step does not work, therefore here
         self.log('lr_rate', self.lr_scheduler.get_lr()[0])
-
         return loss
 
     def validation_step(self, batch, batch_idx):
         # By default logs it per epoch (weighted average over batches)
-        self.log('val_accuracy', self.accuracy(*self.model(batch, mode='val')))
+        out, labels = self.model(batch, mode='val')
+        predictions = self.classifier(out)
+        self.log('val_accuracy', self.accuracy(predictions, labels))
 
     def test_step(self, batch, batch_idx):
         # By default logs it per epoch (weighted average over batches)
-        self.log('test_accuracy', self.accuracy(*self.model(batch, mode='test')))
+        if self.test_val_mode == 'test':
+            out, labels = self.model(batch, mode='test')
+        if self.test_val_mode == 'val':
+            out, labels = self.model(batch, mode='val')
+        predictions = self.classifier(out)
+        self.log('test_accuracy', self.accuracy(predictions, labels))
+    
+    def backward(self, loss, optimizer, optimizer_idx):
+        #override of the backward pass so we can set retain_graph to True
+        loss.backward(retain_graph=True)
```

Comment:
Did you find out why this is needed?

Reply:
Yes. Because in the RobertaGraph, the input is a concatenatation of the document features and the linear projection of the word features. This means that it has a slightly more complicated 'graph' to backpropagate to as it needs to remember that the upper rows of the feature matix came from Roberta (which it doesn't need to update) and the lower rows of the feature matrix came from the linear projection of the word features (which it does need to update). Usualy Pytorch throws away this information to save memory but in this case, it can't throw that information away and we have to let that know by setting retain_graph=True.
This is only used for the RobertaGraph model btw but it doesn't work if I just overwrite the backward method of the RobertaGraph encoder. If this results in memory issues for the other encoders, I can take another look at if we can only do this for the RobertaGraphEncoder.

Reply:
I would prefer to only do this for the RobertaGraph Encoder. can you not simply wrap this with a `iif(isinstance(self.model, RobertaGraphEncoder)){ ... }`?